File: javascript/primelooper.js
function main() {
    process.argv.forEach(function (val, index, array) {
        upperbound = val
    })

    let number_of_primes = getPrimes(upperbound);    
    let cycles = number_of_primes / 78498
    
    console.log("Javascript ran:", cycles, "cycles");
}

function getPrimes(time_limit) {
    let array_of_primes = Array();
    array_of_primes.push(2)
    let current_num = 3
    
    let start = new Date()
    let end = new Date()
    
    while ((end.getTime() - start.getTime()) < time_limit * 1000) {
    
        end = new Date()

        let is_prime = true;
        let sqrtOfI = Math.sqrt(current_num)
        
        for (let j = 0; j < array_of_primes.length; j++) {
            let element = array_of_primes[j];

            if (element > sqrtOfI) {
                break;
            }
            if(current_num % element == 0) {
                is_prime = false;
                break;
            }
        }
        if (is_prime == true) {
            array_of_primes.push(current_num);
        }
        if (current_num >= 1000000) {
            current_num = 3;
        }
        else {
            current_num += 2;
        }
    }
    return array_of_primes.length
}

main();

File: python/primelooper.py
import time, math
import argparse
from multiprocessing import cpu_count, Process, Queue

Q = Queue()

def findprimes(time_limit):
    primes = 0
    procs = []
    for _ in range(cpu_count()):  # Split load evenly across CPU cores
        p = Process(target=calc_primes, args=(time_limit,))
        p.start()
        procs.append(p)

    for _ in procs:
        print(Q.get())
        primes += Q.get()

    return primes


def calc_primes(time_limit):
    start_time = time.time()
    primes = [2]
    current_num = 3
    current_time = time.time()

    while (current_time - start_time) < time_limit:
        current_time = time.time()

        is_prime = True
        sqrt_num = math.sqrt(current_num)

        for _, j in enumerate(primes):
            if j > sqrt_num:
                break

            if current_num % j == 0:
                is_prime = False
                break

        if is_prime:
            primes.append(current_num)

        if current_num >= 1_000_000:
            current_num = 3
        else:
            current_num += 2

    Q.put(len(primes))


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument('time', metavar='N', type=int, nargs='+',
                    help='How long the benchmark goes on')
    args = parser.parse_args()

    primes_found = findprimes(args.time[0])

    print(f"Python ran: {primes_found / 78498} cycle(s)")


if __name__ == "__main__":
    main()